### ISSUE TITLE
Some Suggestions to make Artgen much better

### ISSUE THREAD

**Issue by foxpcteam** [hisorange/artgen#1]
Hi, great project! 
after playing with the Artgen here is what i see that missing.

**1. Simplify creation of new entity.**

I think beside "Data Type" we should have "Field type".
This is much user-friendly in many cased when you want to create new entity with different field types.
We should have "Field type" that will automatically convert to "Data Type".

Field types that we need to have:
* Basic (most of them exist now) - text, number, password, phone, email, url etc...
* Choices - checkbox, radio buttons, single select, multi select, radio group, checkbox group
* Media - internal/external image/video
* Link to (relation will show records from other entity)
* Auto incrementing number with string type(prefix) with option to set specific length. This must have for orders/invoices and etc...
* slate or any other editor.
* Embed field
* Input mask support - include support for masks in the input field. This will allow setting fields such as phone, currency, etc.
* Group of fields - allow creating group of fields that related with repeatable options.

Need to add new options to fields:
Allow to make field as required or not
Allow to make field as read only
Allow to set field min-max length  
Allow to set field as "Personal Data" good for private notes for example
Add tooltip text - this text will be added to the icon near the field
Add Auto-increment
Repeatable fields

Add Dynamic Logic
Conditions making field visible
Conditions making field required / read-only etc...

*** add accessibility to all fields as possible ***

**2. Search function with autocomplete**
We need to have a global search bar in the top menu. We can allow user to select where to search or just to search in all the system

**3. Media Library module with folder structure. (local or external, like s3 and other providers)**

This module will allow a user/team to manage all files from one place with folder structure.

Each folder / file should have permissions

Each folder / file could be assigned / attached embed based on permissions

**4. Internal / External Notifications**
Internal Notifications center can be used to send any type of notifications to user, users or team.

*** Internal Notifications center can be styled as "global notifications" without the option to reply or"social activity" which allow user to replay to each other.

External Notifications:
for example, send notification by email, SMS, telegram, WhatsApp etc... to user on new "task".

**5. Simplify flow + add more options**
Add "clone" button. This will allow using flow as "template".
Add export flow button.
Add a node that will update record.

Basically to simplify the flow we can make "pre-build templates" for most use cases like:
when new record created do something...
When some field updated do something...
etc...

**Comment by hisorange** [comment]
Hehe, I will keep this open.

So, thanks for the throughout analysis. I had similar plans. Mainly I wanna simplify the current version. Since at this phase this is more like a "can this be done?"

Actually the main problem is the complexity, I am planning to cut the down the code into smaller modules and work them out individually. Right now, it's a good base, and verified that the need for a solution like this is there.

Maybe I should open a public RFC channel where those suggestions could be logged.
